### ISSUE TITLE
Linux/Docker Port

### ISSUE THREAD

**Issue by YemotaY** [pi-node/pi-node#3]
Hey I would like to port the pi node to linux and docker. So it's easily runable on raspberrys. 
Which is not only a security aspect bonus but also an energy saver if run on raspberry.
If interest is there hit me up for reference.

Greetings

**Comment by ziyouniaoseo** [comment]
> Hey I would like to port the pi node to linux and docker. So it's easily runable on raspberrys. Which is not only a security aspect bonus but also an energy saver if run on raspberry. If interest is there hit me up for reference.
> 
> Greetings


Pi Node does not have a Linux client, how to port it?

**Comment by hejun041** [comment]
+1

**Comment by capriele** [comment]
+1

**Comment by Anguszax** [comment]
How can I get the sign-in code in my linux system? I already set up all the thing, but I can't find any way to binding my account.

**Comment by wickeds2007** [comment]
Im stuck at sudo apt-get install pi-node, im getting this error <sudo apt-get install pi-node
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package pi-node >
anyone can help me pls?

**Comment by YemotaY** [comment]
I think there's still no support for linux so far?!

**Comment by ziyouniaoseo** [comment]
> 我认为到目前为止还不支持 Linux？！

The Linux version node has been released, official address: https://minepi.com/pi-blockchain/pi-node/linux/

**Comment by wickeds2007** [comment]
I know that, but I get error when I try to download it with that command, sudo apt-get install pi-node, there it's the problem, redirection

**Comment by ziyouniaoseo** [comment]
The Linux version node has been installed and is running normally.




西安智狐
***@***.***

**Comment by wickeds2007** [comment]
How did you get the pi node, tell me the command pls?

**Comment by ziyouniaoseo** [comment]
> How did you get the pi node, tell me the command pls?

Just install it according to the official tutorial. At present, it is only a beta version, and the node binding app function is missing. Windows version of the node migration linux also does not. It is currently in a computationally neutral state, but the data is working well.

**Comment by wickeds2007** [comment]
After terminal command of sudo apt  to install pi node, I receive E: error, cannot find pi-node, so it cannot be installed

**Comment by brianacox** [comment]
> > How did you get the pi node, tell me the command pls?
> 
> Just install it according to the official tutorial. At present, it is only a beta version, and the node binding app function is missing. Windows version of the node migration linux also does not. It is currently in a computationally neutral state, but the data is working well.

Does that mean, you can't actually link it to your phone?   Is that why I can't gain gui (only cli) access, and can't locate the Node Verification Code?

**Comment by wickeds2007** [comment]
No, it's about i cannot install pi-node in raspberry terminal with apt

**Comment by brianacox** [comment]
> How did you get the pi node, tell me the command pls?

Did you work out what to do?   I'm having the same issue...

**Comment by wickeds2007** [comment]
I didn't till now, still waiting to see any extra command, I tried all possibilities from minepi.com, from bitget, from GitHub....none it's working for me

**Comment by brianacox** [comment]
> I didn't till now, still waiting to see any extra command, I tried all possibilities from minepi.com, from bitget, from GitHub....none it's working for me

The documentation is very scant.....   it appears to be running on my MacBook Pro (running Linux) as I've got 6 incoming peers and 8 outgoing peers, but it isolated and not connected to anything- my phone - to do with me.

**Comment by wickeds2007** [comment]
For me the problem it's raspberry pi5 running on Linux and those installation commands, on my PC windows 11 it's working fine

**Comment by wickeds2007** [comment]
![Image](https://github.com/user-attachments/assets/a2ad37d1-1519-43d6-ad22-74dc783c4e06)

**Comment by brianacox** [comment]
> > How did you get the pi node, tell me the command pls?
> 
> Just install it according to the official tutorial. At present, it is only a beta version, and the node binding app function is missing. Windows version of the node migration linux also does not. It is currently in a computationally neutral state, but the data is working well.

Does that mean (sorry non-techy here) that I can't bind it to my phone....?   and this is why there's now Pi Node GUI on Linux Ubunto AND why I can't Locate the Node Verification Code?

**Comment by brianacox** [comment]
> ![Image](https://github.com/user-attachments/assets/a2ad37d1-1519-43d6-ad22-74dc783c4e06)

Where did you get this from?   got the URL?     Is it saying that the Pi Node for Unix only runs on a Raspberry Pi?

**Comment by wickeds2007** [comment]
That screenshot it's made from GitHub copilot, answered to me after i tried to install on raspberry pi5 with help from GitHub copilot, with all errors gained, copy/ pasted from chat with copilot

**Comment by Juhu1982** [comment]
ARM64/aarch64 support for Raspberry Pi 5

Linux support is now available, but Raspberry Pi 5 still appears to be blocked by the CPU architecture.

The current official PiCoreTeam/pi-node-docker Makefile explicitly builds Stellar Core, Horizon, and the final Pi Node image with --platform linux/amd64:
https://github.com/PiCoreTeam/pi-node-docker/blob/master/Makefile

A Raspberry Pi 5 normally runs Linux on aarch64/ARM64, so the official image cannot run natively there at the moment.

Are there plans to provide either:

a native ARM64/aarch64 Pi Node image, or
a multi-architecture Docker image (linux/amd64 + linux/arm64)?

If ARM64 support is currently blocked, it would also be useful to know whether the blocker is Stellar Core, Horizon, another Pi Node component, or simply that an ARM64 build is not yet part of the release pipeline.

I have a Raspberry Pi 5 running Debian 13 ARM64 as an always-on 24/7 server and would be happy to test an official ARM64 beta/development build and provide feedback if that would help.

I am specifically interested in a native ARM64 solution rather than running the existing amd64 image through emulation.

**Comment by YemotaY** [comment]
Since no responce since creation and slightly loosing faith im closing this.
Not willed anymore to help out tho.
For any questions that came up in this ticket please reopen a new one.
Good luck.
